### PR TITLE
Swift Argument Parser for the CLI

### PR DESCRIPTION
**Pull request by liamrosenfeld**
## Summary
Uses the new official Swift Argument Parser for the CLI. Allows for a simpler, more maintainable CLI.

## Review Checklist

- [x] I have responded to reviews and made changes where appropriate.
- [x] I have tested the code
- [x] I have updated comments / documentation related to the changes I made.
- [x] I have rebased my PR branch onto xi-mac/master.

### CODE REVIEW

**Review by nangtrongvuon — APPROVED** [review]
This looks good to me, although the only nit I have with this is I wouldn't use the name `CliHelper` since Swift names are supposed to have initials/abbreviations written in all caps (so `CLIHelper` in this case). Thanks for the PR, this makes the CLI code look much cleaner!

**Comment by liamrosenfeld** [comment]
I renamed them 👍

**Review by jeremywiebe — APPROVED** [review]
Good stuff.  Thanks!

File: Sources/XiCLICore/CLIHelper.swift [hunk 1]
```diff
@@ -12,44 +12,11 @@
...
-            print("waiting for editor to close...")
-            let group = DispatchGroup()
-            group.enter()
-            setObserver(group: group, filePaths: filePaths)
-            group.wait()
-        }
-    }
-    
-    func resolvePath(from input: String) throws -> String {
+struct CLIHelper {
```
[the remark this answers is not in the document]
Comment (cmyr) [reply Sources/XiCLICore/CLIHelper.swift]: I'm happy either way, support both sides, and think that if this has already changed once in review it should stay as is. :)

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 language: rust
-osx_image: xcode10.2
+osx_image: xcode11.3
```
[the remark this answers is not in the document]
Comment (cmyr) [reply .travis.yml]: I was also worried about this but 11.3 is supported by mojave, and I suspect it is required to use the new swift arg parser?

File: Sources/XiCLICore/CLIHelper.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremywiebe) [reply Sources/XiCLICore/CLIHelper.swift]: Fair enough. Like I said, I know it’s subjective so I’m totally fine with leaving it.

[the remark this answers is not in the document]
Comment (jeremywiebe) [reply Sources/XiCLICore/CLIHelper.swift]: Fair enough. Like I said, I know it’s subjective so I’m totally fine with leaving it.

File: .travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (liamrosenfeld) [reply .travis.yml]: Yeah, in order to use SPM dependencies in a application it needs to use Xcode 11

File: Sources/XiCLICore/CLIHelper.swift [hunk 3]
```diff
@@ -66,28 +33,28 @@ public final class CommandLineTool {
         let pathString = canonicalPath(input)
 
         guard pathIsNotDirectory(pathString) else {
-            throw CliError.pathIsDirectory
+            throw ValidationError("The path entered is to a directory")
```
[the remark this answers is not in the document]
Comment (cmyr) [reply Sources/XiCLICore/CLIHelper.swift]: @liamrosenfeld this looks like the only unaddressed thing? It sounds like a reasonable addition, then this looks good to go.

[the remark this answers is not in the document]
Comment (liamrosenfeld) [reply Sources/XiCLICore/CLIHelper.swift]: Implemented 👍

**Comment by cmyr** [comment]
Thanks!

**Comment by liamrosenfeld** [comment]
Thanks for a great review process!

**Comment by jeremywiebe** [comment]
Yay!  Thanks @liamrosenfeld

**Pull request merged**